### ISSUE TITLE
precompile-safety

### ISSUE THREAD

**Issue by stevengj** [bnewbold/PyX.jl#1]
To be precompile-safe, you should really initialize all your global Python variables like `canvas = pywrap(pyimport("pyx.canvas"))` in your `__init__` function.

Also, you should probably add `pyimport_conda("pyx", "pyx", "bioconda")` at the beginning of `__init__` so that the `pyx` module can be automatically installed of PyCall is configured to use Conda.  (This will require PyCall 1.6, to be tagged shortly: JuliaLang/METADATA.jl#5201.)

See https://github.com/stevengj/PyCall.jl#using-pycall-from-julia-modules

**Comment by bnewbold** [comment]
Thanks for pointing out the precompile-safe issue! I'll take a look.

Regarding the bioconda route for installing the python PyX module, have you tested that this would actually work on a machine without a system-wide TeX install of some sort? I haven't looked carefully, but glancing at the bioconda recipe (https://github.com/bioconda/bioconda-recipes/tree/master/recipes/pyx), I don't see how it installs or depends on TeX, which is the difficult dependency for cross-platform installation. Eg, AFAIK there is no stable solution for automated installation of TeX on OS X; homebrew gave up and the "cask" I found didn't work reliably.

**Comment by stevengj** [comment]
No, I haven't tried it; I didn't realize there was a LaTeX dependency

**Comment by bnewbold** [comment]
As mentioned in the README, switching PyCall to python3 is also necessary to use the latest version of PyX, which increases the complexity of install automation.

**Comment by bnewbold** [comment]
Moved auto-install discussion to it's own issue (#3).

**Comment by bnewbold** [comment]
Hey @stevengj, I just took a whack at this. Not withstanding the failure on 0.5 (nightly), tests seem to pass.
Could you take a look and see if what I did was sane? The commit was 29af2e61e6f511.
